native/cocos/2d/renderer/Batcher2d.cpp

```diff
@@ -94,9 +95,18 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
 }
 
 void Batcher2d::fillBuffersAndMergeBatches() {
+    size_t index = 0;
     for (auto* rootNode : _rootNodeArr) {
+        // _batches will add by generateBatch
```

Comment:
It is better to add this comment above the code `generateBatch(_currEntity, _currDrawInfo)`.